components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -3584,334 +3576,608 @@ end subroutine mam_soaexch_1subarea
-                  if (lstooa .gt. 0) then
-                    write(lund,'(a,i4,2(2x,a),1p,10e14.6)') 'RENAME qdels', iq,   &
-                        cnst_name(lsfrma+loffset), cnst_name(lstooa+loffset),   &
-                        deltat*dqdt(i,k,lsfrma), deltat*(dqdt(i,k,lsfrma) - xfertend),   &
-                        deltat*dqdt(i,k,lstooa), deltat*(dqdt(i,k,lstooa) + xfertend)
-                  else
-                    write(lund,'(a,i4,2(2x,a),1p,10e14.6)') 'RENAME qdels', iq,   &
-                        cnst_name(lsfrma+loffset), cnst_name(lstooa+loffset),   &
-                        deltat*dqdt(i,k,lsfrma), deltat*(dqdt(i,k,lsfrma) - xfertend)
-                  end if
-                end if
-                end if
-!   diagnostic output end   ------------------------------------------
-#endif
+           fmode_dist_tail_fac(src_mode) = sqrt_half/alnsg_aer(src_mode) !factor for computing distribution tails of the  "src mode"
 
+           !compute volume to number high and low limits with relaxation coefficients (watch out for repeated calculations)
+           v2n_lo_rlx(src_mode) = compute_vol_to_num_ratio(src_mode, dgnumlo_aer) * frelax
+           v2n_hi_rlx(src_mode) = compute_vol_to_num_ratio(src_mode, dgnumhi_aer) / frelax
 
-      end do mainloop1_ipair
+           !A factor for computing diameter at the tails of the distribution
+           ln_diameter_tail_fac(src_mode) = 3.0 * (alnsg_aer(src_mode)**2)
 
+           !Cut-off (based on geometric mean) for making decision to do inter-mode transfers
+           diameter_cutoff(src_mode) = sqrt(   &
+                dgnum_aer(src_mode)*exp(1.5*(alnsg_aer(src_mode)**2)) *   &
+                dgnum_aer(dest_mode)*exp(1.5*(alnsg_aer(dest_mode)**2)) )
 
-      return
-      end subroutine mam_rename_1subarea
+           ln_dia_cutoff(src_mode) = log(diameter_cutoff(src_mode)) !log of cutt-off
+           diameter_threshold(src_mode) = 0.99*diameter_cutoff(src_mode) !99% of the cutoff
 
+        enddo
 
-!----------------------------------------------------------------------
-!----------------------------------------------------------------------
+      end subroutine find_renaming_pairs
+      !----------------------------------------------------------------------
+      !----------------------------------------------------------------------
+
+      subroutine compute_size_factor(imode, size_factor)
+        ! Compute size factor for a mode
+        implicit none
+
+        integer,  intent(in) :: imode     !mode number
+        real(r8), intent(inout) :: size_factor(:) !size factor
+
+        size_factor(imode) = (pi/6.)*exp(4.5*(alnsg_aer(imode)**2))
+
+      end subroutine compute_size_factor
+
+      !----------------------------------------------------------------------
+      !----------------------------------------------------------------------
+
+      pure function compute_vol_to_num_ratio(imode, diameter) result(v2n)
```

Comment:
Something for us to consider in our Fortran style: in mathematics, we often treat functions as "nouns", whereas algorithms (subroutines) are often described as "verbs". `compute_vol_to_num_ratio` is currently a verb, and if we wanted to refer to it as a noun, we might rename it to `vol_to_num_ratio`. Not a big deal, but it might read easier within expressions.